lib/freebsd.ex

```diff
@@ -36,6 +38,78 @@ defmodule FreeBSD do
 
   defp freebsd_config, do: Mix.Project.config() |> Keyword.fetch!(:freebsd)
 
+  defp freebsd_config(key),
+    do: Mix.Project.config() |> Keyword.fetch!(:freebsd) |> Map.get(key)
+
   defp with_deps(manifest, nil), do: manifest
   defp with_deps(manifest, deps), do: Map.put(manifest, :deps, deps)
+
+  defp with_user(manifest, nil), do: manifest
+  defp with_user(manifest, username), do: Map.put(manifest, :users, [username])
+
+  def with_scripts(manifest, username) do
+    manifest
+    |> Map.put(:scripts, %{
+      "pre-install" =>
+        Enum.join(
+          [
+            create_user_script(username),
+            create_config_dir_script()
+          ],
+          "\n"
+        )
+    })
+  end
+
+  defp create_user_script(nil), do: ""
+
+  defp create_user_script(username) do
```

Comment:
Is this pretty much copied from postgres? I'll have to experiment with it a bit.

Interesting that the scripts look like they're inlined into the manifest, instead of being files on disk.

I think it will be cleaner to extract this to an EEx template and eval it for the result.

Reply:
They're definitely inlined into the manifest in the postgres server pkg. I think it makes sense in the context of `pre-install` and `post-deinstall`, both of which run without any of the packages files available.

Reply:
This is what I'm using to look at other manifest examples, as I have not yet found documentation of the manifest schema.

```
sudo pkg fetch postgresql15-server
pkg info -R -F /var/cache/pkg/postgresql15-server-15.b2.pkg
```